java/src/jmri/jmrit/logixng/actions/SimulateTurnoutFeedback.java

```diff
@@ -125,32 +125,34 @@ private void addTurnoutListener(Turnout turnout) {
         if (!_turnouts.containsKey(turnout.getSystemName())) {
             TurnoutInfo ti = new TurnoutInfo(turnout);
             _turnouts.put(turnout.getSystemName(), ti);
-            turnout.addPropertyChangeListener("feedbackchange", this);
-            turnout.addPropertyChangeListener("turnoutFeedbackFirstSensorChange", this);
-            turnout.addPropertyChangeListener("turnoutFeedbackSecondSensorChange", this);
+            turnout.addPropertyChangeListener(Turnout.PROPERTY_FEEDBACK_MODE, this);
+            turnout.addPropertyChangeListener(Turnout.PROPERTY_TURNOUT_FEEDBACK_FIRST_SENSOR, this);
+            turnout.addPropertyChangeListener(Turnout.PROPERTY_TURNOUT_FEEDBACK_SECOND_SENSOR, this);
             if (hasTurnoutFeedback(turnout)) {
-                turnout.addPropertyChangeListener("CommandedState", _turnoutListener);
+                turnout.addPropertyChangeListener(Turnout.PROPERTY_COMMANDED_STATE, _turnoutListener);
                 ti._hasListener = true;
             }
         }
     }
 
     private void removeTurnoutListener(Turnout turnout) {
         TurnoutInfo ti = _turnouts.remove(turnout.getSystemName());
-        turnout.removePropertyChangeListener("feedbackchange", this);
-        turnout.removePropertyChangeListener("turnoutFeedbackFirstSensorChange", this);
-        turnout.removePropertyChangeListener("turnoutFeedbackSecondSensorChange", this);
+        turnout.removePropertyChangeListener(Turnout.PROPERTY_FEEDBACK_MODE, this);
+        turnout.removePropertyChangeListener(Turnout.PROPERTY_TURNOUT_FEEDBACK_FIRST_SENSOR, this);
+        turnout.removePropertyChangeListener(Turnout.PROPERTY_TURNOUT_FEEDBACK_SECOND_SENSOR, this);
         if (ti != null && ti._hasListener) {
-            turnout.removePropertyChangeListener("CommandedState", _turnoutListener);
+            turnout.removePropertyChangeListener(Turnout.PROPERTY_COMMANDED_STATE, _turnoutListener);
             ti._hasListener = false;
         }
     }
 
     /** {@inheritDoc} */
     @Override
     public synchronized void propertyChange(PropertyChangeEvent evt) {
-        if (evt.getPropertyName().equals("beans")) {
-            if (!(evt.getSource() instanceof TurnoutManager)) return;
+        if (Manager.PROPERTY_BEANS.equals(evt.getPropertyName())) {
+            if (!(evt.getSource() instanceof TurnoutManager)) {
+                return;
```

Comment:
It would be good to log an error here if the source is not a TurnoutManager.
It shouldn't happen.

Reply:
No probs, have updated this PR.